lib/utils/import/googlekeep/index.ts

```diff
@@ -0,0 +1,115 @@
+import { EventEmitter } from 'events';
+import { endsWith, isEmpty, get, has } from 'lodash';
+import JSZip from 'jszip';
+
+import CoreImporter from '../';
+
+let fs = null;
+const isElectron = has(window, 'process.type');
+if (isElectron) {
+  fs = __non_webpack_require__('fs'); // eslint-disable-line no-undef
+}
+
+class GoogleKeepImporter extends EventEmitter {
+  constructor({ noteBucket, tagBucket, options }) {
+    super();
+    this.noteBucket = noteBucket;
+    this.tagBucket = tagBucket;
+    this.options = options;
+  }
+
+  importNotes = filesArray => {
+    if (isEmpty(filesArray)) {
+      this.emit('status', 'error', 'No file to import.');
+      return;
+    }
+
+    const coreImporter = new CoreImporter({
+      noteBucket: this.noteBucket,
+      tagBucket: this.tagBucket,
+    });
+
+    let importedNoteCount = 0;
+
+    const importJsonString = jsonString => {
+      // note: If importing the note fails, it is silently ignored by the
+      // promise below. This is okay since the warning message would be hidden
+      // by the next progress update anyway.
+      const importedNote = JSON.parse(jsonString);
+
+      if (
+        !importedNote.title &&
+        !importedNote.textContent &&
+        !importedNote.listContent
+      ) {
+        // empty note, skip
+        return;
+      }
+
+      const title = importedNote.title;
+
+      const importedContent = importedNote.listContent
+        ? importedNote.listContent // Note has checkboxes, no text content
+            .map(item => `- [${item.isChecked ? 'x' : ' '}] ${item.text}`)
+            .join('\n')
+        : importedNote.textContent;
```

Comment:
interesting. a note can only _either_ be a list _or_ some text?

Reply:
Yeah, in Keep, you can "show checkboxes", which turns the whole note into a checklist, and as far as I can tell there's no way to have regular text then.